### PR TITLE
Fix torch backend symmetric pad returning edge padding

### PR DESCRIPTION
**Pull request by CodingSelim**
## Description

`keras.ops.pad(..., mode="symmetric")` on the torch backend was returning edge/replicate padding instead of symmetric padding. The torch backend aliased `mode="symmetric"` to torch's `"replicate"` mode, which just repeats the edge value. That is numpy's `"edge"` mode, not `"symmetric"`.

Example on the torch backend before this change:

```python
>>> keras.ops.pad([1,2,3,4,5], ((2,3),), mode="symmetric")
[1, 1, 1, 2, 3, 4, 5, 5, 5, 5]   # wrong (edge padding)
```

numpy (and the jax/tf backends) give:

```python
>>> np.pad([1,2,3,4,5], (2,3), mode="symmetric")
[2, 1, 1, 2, 3, 4, 5, 5, 4, 3]   # correct mirror
```

Fix mirrors the values manually per axis (torch has no native symmetric mode). The index math also handles pad widths larger than the axis size, matching numpy.

The existing test only padded an all-ones array, where symmetric and edge padding are identical, so it never caught this. Updated the 2D case to use varied values.

Verified against `numpy.pad` across 1D/2D/3D, per-dim and broadcast pad widths, larger-than-size pads, and int/float dtypes. `test_pad` passes on the torch backend and fails without the source fix.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

**Review by hertschuh — APPROVED** [review]
Thank you for the fix!
